### PR TITLE
Hotfix for rebinning

### PR DESCRIPTION
**Pull request by rcooke-ast**
This PR offers a hotfix for rebinning. It cleans up the ugly evlist version of rebinning (with a speedup - see timings below) and removes the old "rebin" function which is really just slicing, and not rebinning (i.e. the old version doesn't conserve flux).

image

### CODE REVIEW

**Review by jhennawi — APPROVED** [review]
Looks good @rcooke-ast. Thanks. I think we should probably still run tests.

File: pypeit/utils.py [hunk 1]
```diff
@@ -501,20 +502,36 @@ def rebin(a, newshape):
...
 
+    Args:
+        img (`numpy.ndarray`_):
+            A 2D input image
+        shape (:obj:`tuple`):
+            The desired shape to be returned. The elements of img.shape
+            should be an integer multiple of the elements of shape.
+
+    Returns:
+        img_out (`numpy.ndarray`_): The input image rebinned to shape
```
Comment (profxj) [inline pypeit/utils.py]: I think the syntax is without naming the variable, at least left of the `:`

**Review by profxj — APPROVED** [review]
All good;  one minor doc edit

File: pypeit/utils.py (hunk 1 above)
Comment (rcooke-ast) [reply pypeit/utils.py]: Quite right... I'll update this!

**Comment by rcooke-ast** [comment]
All tests are passing...
image

**Pull request merged**